js/screens/authentication/IdpLoginScreen.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+
+import * as React from 'react'
+import { connect } from 'react-redux'
+import {
+  type NavigationScreenProp,
+  type NavigationState
+} from 'react-navigation'
+import { WebView } from 'react-native'
+import { Container, Header, Left, Button, Icon, Text, Body } from 'native-base'
+
+import { type ReduxProps } from '../../actions/types'
+import { type GlobalState } from '../../reducers/types'
+import { type SessionState } from '../../store/reducers/session'
+import config from '../../config'
+import I18n from '../../i18n'
+import { extractLoginResult } from '../../api'
+import { loginSuccess, loginFailure } from '../../store/actions/session'
+
+type ReduxMappedProps = {
+  session: SessionState
+}
+
+type OwnProps = {
+  navigation: NavigationScreenProp<NavigationState>
+}
+
+type Props = ReduxMappedProps & ReduxProps & OwnProps
+
+const LOGIN_BASE_URL = `${config.apiUrlPrefix}/login?entityID=`
+
+/**
+ * A screen that allow the user to login with an IDP.
+ * The IDP page is opened in a WebView
+ */
+class IdpLoginScreen extends React.Component<Props> {
+  render(): React.Node {
+    const { session } = this.props
```

Comment:
I think this component should expect a `session` in the right state (let flow check that `session` is the right type that has a defined `idp`), the next `if` statement shouldn't be necessary

Reply:
`session` is taken from the store in mapStateToProps, i don't think we can be sure of the exact SessionState subtype.

Reply:
my point is that checking whether the type passed to a function is the right one should be a responsibility of the caller, not the function itself (defensive programming)